src/KIM/kim_init.cpp

```diff
@@ -157,20 +153,53 @@ void get_kim_unit_names(
     chargeUnit = KIM_CHARGE_UNIT_e;
     temperatureUnit = KIM_TEMPERATURE_UNIT_K;
     timeUnit = KIM_TIME_UNIT_fs;
-  } else if ((system_str == "lj") ||
-             (system_str == "micro") ||
-             (system_str == "nano")) {
-    error->all(FLERR, "LAMMPS unit_style {} not supported "
-                                  "by KIM models", system_str);
+  } else if ((system_str == "lj") || (system_str == "micro") || (system_str == "nano")) {
+    error->all(FLERR, "LAMMPS unit_style {} not supported by KIM models", system_str);
   } else {
     error->all(FLERR, "Unknown unit_style");
   }
 }
-}  // namespace
+}    // namespace
 
-void KimInit::determine_model_type_and_units(char * model_name,
-                                             char * user_units,
-                                             char ** model_units,
+void KimInit::print_dirs(struct KIM_Collections *const collections) const
+{
+  int kim_error = 0;
+  int dirListExtent = 0;
+  input->write_echo("#=== KIM is looking for 'Portable Models' in these directories ===\n");
+  std::vector<struct KIM_Collection> collection_list;
+  collection_list.push_back(KIM_COLLECTION_currentWorkingDirectory);
+  collection_list.push_back(KIM_COLLECTION_environmentVariable);
+  collection_list.push_back(KIM_COLLECTION_user);
+  collection_list.push_back(KIM_COLLECTION_system);
+
+  for (auto col : collection_list) {
+    kim_error = KIM_Collections_CacheListOfDirectoryNames(
+        collections, col, KIM_COLLECTION_ITEM_TYPE_portableModel, &dirListExtent);
+    if (!kim_error) {
+      for (int i = 0; i < dirListExtent; ++i) {
+        char const *name;
+        kim_error = KIM_Collections_GetDirectoryName(collections, i, &name);
```

Comment:
@ellio167 when I try this feature on a Fedora 34 machine with the kim-api rpm package, kim_error is 1 and thus no directories are printed.

Reply:
Ah, right; I found a bug in the kim-api as I was putting this together...  It's fixed the the devel branch of kim-api, but not going to be released for a while.  I'll push a temporary fix.

Reply:
@ellio167 I don't like that kind of commented out code much. I've updated the code with a solution that feels cleaner to me.

Reply:
For the future, you may want to consider having a "comparable" version tag (like I created here locally) included in the KIM API and then you can use that to make the KIM package code backward compatible to a range of API versions.
Unless you want to force people to always upgrade immediately, that is.

Reply:
Looks good. Thanks.